torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -7146,6 +7146,402 @@ Tensor values_backward(const Tensor& grad, const Tensor& self) {
+  };
+
+  auto __continued_fraction_evaluation = [&](const Tensor& should_stop,
+                                             int32_t iteration,
+                                             const std::vector<Tensor>& values,
+                                             const std::vector<Tensor>& gradients) -> std::tuple<Tensor,
+                                                                                                 int32_t,
+                                                                                                 std::vector<Tensor>,
+                                                                                                 std::vector<Tensor>> {
+    std::tuple<std::vector<Tensor>, std::vector<Tensor>, Tensor> _ret = __continued_fraction_step(iteration, values, gradients, _betainc_even_partial_numerator);
+    std::vector<Tensor>& _new_values = std::get<0>(_ret);
+    std::vector<Tensor>& _new_gradients = std::get<1>(_ret);
+    //std::vector<Tensor>& _ = std::get<2>(_ret);
+
+    std::tuple <std::vector<Tensor>, std::vector<Tensor>, Tensor> ret = __continued_fraction_step(iteration, _new_values, _new_gradients, _betainc_odd_partial_numerator);
+    std::vector<Tensor>& new_values = std::get<0>(ret);
+    std::vector<Tensor>& new_gradients = std::get<1>(ret);
+    const Tensor& delta = std::get<2>(ret);
+    Tensor stop = should_stop | (at::abs(delta-1.0) < tolerance);
+    return std::make_tuple(std::move(stop), iteration + 1, std::move(new_values), std::move(new_gradients));
+
+  };
+
+  Tensor apb = _a + _b;
+  Tensor ap1 = _a + 1.0;
+  // Initialization and first step of modified Lentz's method.
+  Tensor initial_ratio_numerators = at::ones_like(_x);
+  Tensor initial_ratio_denominators = 1.0 - apb * _x / ap1;
+  initial_ratio_denominators = at::where(at::abs(initial_ratio_denominators) < small, small, initial_ratio_denominators);
+  initial_ratio_denominators = at::reciprocal(initial_ratio_denominators);
+  Tensor initial_convergent = initial_ratio_denominators;
+  std::vector<Tensor> values = {std::move(initial_ratio_numerators), std::move(initial_ratio_denominators), std::move(initial_convergent)};
+
+  Tensor initial_dratio_denominators = at::cat({1.0 - _b, ap1}, -1) * _x / at::square(_x * apb - ap1);
+  Tensor initial_dratio_numerators = at::zeros_like(initial_dratio_denominators);
+  Tensor initial_dconvergent = initial_dratio_denominators;
+  std::vector<Tensor> gradients = {std::move(initial_dratio_numerators), std::move(initial_dratio_denominators), std::move(initial_dconvergent)};
+
+  Tensor stop = ~_use_continued_fraction;
+
+  for(int32_t i = 0; i < max_iterations; i++) {
+    std::tuple<Tensor, int32_t, std::vector<Tensor>, std::vector<Tensor>> ret = __continued_fraction_evaluation(stop, i + 1, values, gradients);
+    stop = std::get<0>(ret);
+    values = std::get<2>(ret);
+    gradients = std::get<3>(ret);
+    if (stop.all().item<bool>()) //TODO: It can be bottleneck..
```

Comment:
Any Suggestion for Performance Improvement?

Reply:
On GPU it might be more performant to always do the 100 iterations here? Maybe we could do more fancy async checking of these conditions, but I would definitely keep that for a follow up PR.
For CPU, this will be good for sure.